.github/workflows/ci.yml

```diff
@@ -1,180 +1,228 @@
+        )
+      )
     name: Rust Format
     runs-on: ubuntu-latest
     steps:
       - name: Checkout the latest code
         id: git_checkout
-        uses: actions/checkout@v3
-      - name: Define Rust Toolchain
-        id: define_rust_toolchain
-        run: echo "RUST_TOOLCHAIN=$(cat ./rust-toolchain)" >> $GITHUB_ENV
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
       - name: Setup Rust Toolchain
         id: setup_rust_toolchain
-        uses: actions-rust-lang/setup-rust-toolchain@v1
+        uses: actions-rust-lang/setup-rust-toolchain@f3c84ee10bf5a86e7a5d607d487bf17d57670965 # v1.5.0
         with:
-          toolchain: ${{ env.RUST_TOOLCHAIN }}
           components: rustfmt
-      - name: Rustfmt 
-        id: rustfmt
-        uses: actions-rust-lang/rustfmt@v1
-
-  ## Release tests: Execute on every run
-  release-tests:
-    name: Release Tests
-    uses: stacks-network/stacks-blockchain/.github/workflows/stacks-blockchain-tests.yml@master
+          cache: false
 
-  ## Checked for leaked credentials: Execute on every run
-  leaked-cred-test:
-    name: Leaked Credential Test
-    runs-on: ubuntu-latest
-    steps:
-      - name: Extract branch name
-        id: extract_branch
-        if: ${{ github.event_name != 'pull_request' }}
-        run: echo "BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/})" >> $GITHUB_ENV
-      - name: Extract branch name
-        id: extract_branch_pr
-        if: ${{ github.event_name == 'pull_request' }}
-        run: echo "BRANCH_NAME=$(echo ${GITHUB_HEAD_REF})" >> $GITHUB_ENV
-      - name: Branch name
-        run: echo running on branch ${{ env.BRANCH_NAME }}
-      - name: Checkout the latest code
-        id: git_checkout
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - name: TruffleHog Scan
-        id: trufflehog_check
-        uses: trufflesecurity/trufflehog@main
-        with:
-          path: ./
-          base: ${{ env.BRANCH_NAME }}
-          head: HEAD
+      - name: Rustfmt
+        id: rustfmt
+        uses: actions-rust-lang/rustfmt@2d1d4e9f72379428552fa1def0b898733fb8472d # v1.1.0
 
-  ###############################################
-  ## Build Tagged Release
-  ###############################################
-  ## Build source binaries
-  ## Only run if:
-  ##    - Tag is provided
-  ##    - OR
-  ##        - Not the default branch
-  ##        - AND
-  ##        - Not a PR
-  build-source:
-    if: ${{ inputs.tag != '' || (github.ref != format('refs/heads/{0}', github.event.repository.default_branch) && !contains(github.ref, 'refs/pull')) }}
-    name: Build Binaries
-    uses: stacks-network/stacks-blockchain/.github/workflows/build-source-binary.yml@master
+  ######################################################################################
+  ## Create a tagged github release
+  ##
+  ## Runs when the following is true:
+  ##  - tag is provided
```

Comment:
There's a lot of logic around tags, but I don't understand the significance of the tags (this could be a me issue). In addition to describing these conditions as dependent on `tag`s being provided what do you think of describing these conditions with what the tag's significance is to the job?

Reply:
this is done for historic reasons mainly - when we have a release, it's tagged with a version. 
in our repo/CI - we have an input for the ci.yml workflow that creates the tag vs tagging a commit and pushing it (which would also work, but historically we've used the input tag). 

the tag is created via the `github-releese.yml` workflow (implicitly) creating the tag when a github release is created. 

it may be tribal knowledge that i can comment better, but if you don't input a tag...no release is created and subset of the CI tests are run. if you do add an input tag, all the tests are run and release is made (with the binary archives/docker images)

Reply:
Yeah, do you think adding a comment would be okay? Tribal knowledge isn't very bus proof.

Reply:
I'm not sure if adding more comments would be useful outside of what already exists. 
there already exists some docs covering how tags are used, and the comments in `ci.yml` and `github-release.yml` should be enough to fill in the gaps without being overly verbose. if you think there's something extra that can be done here, can you provide an example? i'm simply not seeing how it would improve the understanding. 

https://github.com/stacks-network/stacks-blockchain/blob/master/docs/ci-release.md
https://github.com/stacks-network/stacks-blockchain/blob/master/docs/release-process.md